File: Java/Stack/DoubleStack.java

import java.util.Arrays;

public class DoubleStack {
    static int stackValueLeft = -1;
    static int stackArray[] = new int[100];
    static int stackValueRight = stackArray.length;

    public void showStackArray() {
        System.out.println(Arrays.toString(stackArray));
    }

    public void showStackIndex() {
        System.out.println(stackValueLeft);
        System.out.println(stackValueRight);
    }

    private void pushfromLeft(int number) {
        if (stackValueLeft < stackValueRight) {
            ++stackValueLeft;
            stackArray[stackValueLeft] = number;
        } else {
            System.out.println("No space.");
        }
    }

    private void pushfromRight(int number) {
        if (stackValueRight > stackValueLeft) {
            --stackValueRight;
            stackArray[stackValueRight] = number;
        } else {
            System.out.println("No space.");

        }
    }

    private void popOfLeft() {
        if (stackValueLeft >= 0) {
            --stackValueLeft;
            stackArray[stackValueLeft + 1] = 0;
        } else {
            System.out.println("No value in the stack.");
        }
    }

    private void popOfRight() {
        if (stackValueRight < 100) {
            stackArray[stackValueRight] = 0;
            ++stackValueRight;
        } else {
            System.out.println("No value in the stack.");
        }
    }

    public static void main(String[] args) {
        DoubleStack stack = new DoubleStack();
        stack.pushfromLeft(5);
        stack.showStackArray();
        stack.showStackIndex();
        stack.pushfromRight(4);
        stack.showStackArray();
        stack.showStackIndex();
        stack.pushfromRight(3);
        stack.showStackArray();
        stack.showStackIndex();
        stack.pushfromRight(2);
        stack.showStackArray();
        stack.showStackIndex();
        stack.popOfLeft();
        stack.showStackArray();
        stack.showStackIndex();
        stack.popOfRight();
        stack.showStackArray();
        stack.showStackIndex();
        stack.popOfRight();
        stack.showStackArray();
        stack.showStackIndex();
        stack.popOfRight();
        stack.showStackArray();
        stack.showStackIndex();

    }
}

File: Python/Stack/DoubleStack.py
class Stack:
    def __init__(self, quantity):
        self.quantity = quantity
        self.topOfLeft = -1
        self.topOfRight = self.quantity - 1
        self.stackArray = [0] * self.quantity

    def showStackArray(self):
        print(self.stackArray)

    def showStackIndex(self):
        print(f"Left index is {self.topOfLeft} and right index is {self.topOfRight}")

    def pushFromLeft(self, value):
        if self.topOfLeft < self.topOfRight:
            self.topOfLeft += 1
            self.stackArray[self.topOfLeft] = value
        else:
            print("Stack is full.")

    def pushFromRight(self, value):
        if self.topOfRight > self.topOfLeft:
            self.topOfRight -= 1
            self.stackArray[self.topOfRight] = value
        else:
            print("The stack is full.")

    def popFromLeft(self):
        if self.topOfLeft >= 0:
            self.stackArray[self.topOfLeft] = 0
            self.topOfLeft -= 1
        else:
            print("Stack is empty from left.")

    def popFromRight(self):
        if self.topOfRight < self.quantity - 1:
            self.stackArray[self.topOfRight] = 0
            self.topOfRight += 1
        else:
            print("Stack is empty from right.")

    def isEmptyFromLeft(self):
        return self.topOfLeft == -1

    def isEmptyFromRight(self):
        return self.topOfRight == self.quantity - 1


# Example usage
stack = Stack(10)
stack.showStackIndex()
stack.pushFromLeft(1)
stack.pushFromLeft(2)
stack.pushFromLeft(3)
stack.pushFromLeft(4)
stack.pushFromLeft(5)
stack.pushFromLeft(6)
stack.pushFromRight(7)
stack.pushFromRight(8)
stack.pushFromRight(9)
stack.pushFromRight(10)
stack.pushFromRight(11)  # This should print "The stack is full."
stack.showStackIndex()
stack.showStackArray()
